### ISSUE TITLE
`semi-spacing`: no support in types and interfaces

### ISSUE THREAD

**Issue by Knutas** [eslint-stylistic/eslint-stylistic#680]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

There is no `ts/semi-spacing` to add support in types and interfaces for spaces before and after semicolon.

Example:
```TypeScript
interface Foo {
	name: string;greet: () => string ;
}

type Union = number | string ;
```

The interface is missing a space after the first semicolon and has an unexpected space before the last semicolon.
The type has an unexpected space before the semicolon.

### Reproduction

^

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
